samples/valueentity-counter-with-action-controller/pom.xml

```diff
@@ -0,0 +1,349 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.example</groupId>
+  <artifactId>valueentity-counter</artifactId>
```

Comment:
Do we need a full new sample for this? can't it be added to the existing `valueentity-counter` example? The only thing we might need to "fake" for the docs is the registration in Main.

Reply:
Exactly, that's where I got blocked and what triggered me to copy the sample. 

I see a few options: 
* fake the registration in the value entity docs
* copy like here - I could have copied less code, just enough to make it compile. 
* copy and add a script to keep things in sync, like we do in the platform guide

Another option is change the registration. Instead of one method `withComponents`, we could have a method per component eg: `registerCounter`, `registerDoubleCounterAction`. 

We will lose the the compile time check in the main. When adding a new component, nothing will tell the user that they need to go there fix it. 

But I also realised that the `withComponents` may be annoying in some situations. For instance, if you need to temporarily disable a component, you will need to remove the proto file and all generated classes. That can be very annoying during development cycle.

Reply:
I found a little hack that reduce the duplication by a lot